modules/apps/kate/default.nix

```diff
@@ -0,0 +1,192 @@
+        type = lib.types.bool;
+      };
+
+      indent.tabFromEverywhere = lib.mkOption {
+        description = "Whether the tabulator key increases intendation independent from the current cursor position.";
+        default = false;
+        type = lib.types.bool;
+      };
+
+      indent.undoByShiftTab = lib.mkOption {
+        description = "Whether to unindent the current line by one level with the shortcut Shift+Tab";
+        default = true;
+        type = lib.types.bool;
+      };
+    };
+  };
+
+  config.assertions = [
+    {
+      assertion = cfg.editor.indent.undoByShiftTab || (!cfg.editor.indent.tabFromEverywhere);
+      message = "Kate does not support both 'undoByShiftTab' to be disabled and 'tabFromEverywhere' to be enabled at the same time.";
+    }
+  ];
+
+  config.programs.plasma.configFile."katerc" = lib.mkIf cfg.enable {
+    "KTextEditor Document" = {
+      "Auto Detect Indent" = cfg.editor.indent.autodetect;
+      "Indentation Width" = cfg.editor.indent.width;
+      "Tab Handling" = (tabHandlingMode cfg.editor.indent);
+      "Tab Width" = cfg.editor.tabWidth;
+      "Keep Extra Spaces" = cfg.editor.indent.keepExtraSpaces;
+      "ReplaceTabsDyn" = cfg.editor.indent.replaceWithSpaces;
+    };
+
+    "KTextEditor Renderer" = {
+      "Show Indentation Lines" = cfg.editor.indent.showLines;
+
+
+      # COLORTHEME (cannot define this below)
+      # Do pick the theme if the user chose one,
+      # Do not touch the theme settings otherwise
+      "Auto Color Theme Selection" = lib.mkIf (cfg.editor.theme.name != "") false;
+      "Color Theme" = lib.mkIf (cfg.editor.theme.name != "") cfg.editor.theme.name;
+    };
+  };
+
+
+  # ==================================
+  #     COLORTHEME
+
+  options.programs.kate.editor.theme = {
+    src = lib.mkOption {
+      description = ''
+        The path of a theme file for the KDE editor (not the window color scheme).
+        Obtain a custom one by using the GUI settings in kate. If you want to use a system-wide
+        editor color scheme set this path to null. If you set the metadata.name entry in the file
+        to a value that matches the name of a system-wide color scheme undesired behaviour may
+        occur. The activation will fail if a theme with the filename `<name of your theme>.theme`
+        already exists.'';
+        type = lib.types.nullOr lib.types.path;
+      default = null;
+    };
+
+    name = lib.mkOption {
+      description = ''
+        The name of the theme in use. May be a system theme.
+        If a theme file was submitted this setting will be set automatically.
+      '';
+      type = lib.types.str;
+      default = "";
+    };
+  };
+
+  config.programs.kate.editor.theme = {
+    name = lib.mkIf (cfg.enable && null != cfg.editor.theme.src) (lib.mkForce (builtins.fromJSON (builtins.readFile cfg.editor.theme.src))."metadata"."name");
+    # kate's naming scheme is ${themename}.theme
+    # which is why we use the same naming scheme here
```

Comment:
Again I would prefer comments above

Reply:
"above" as in "the comment line before the code line"?
Done